whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy

```diff
@@ -633,13 +633,8 @@ class PostgreSQLComponent implements Storage {
             rigUpdateStatement(updateStatement, doc, modTime, changedIn, changedBy, collection, deleted)
             updateStatement.execute()
 
-            // The versions and identifiers tables are NOT under lock. Synchronization is only maintained on the main table.
             saveVersion(doc, connection, createdTime, modTime, changedIn, changedBy, collection, deleted)
             refreshDerivativeTables(doc, connection, deleted)
-            for (Tuple2<String, String> depender : getDependers(doc.getShortId())) {
-                updateMinMaxDepModified((String) depender.get(0), connection)
-                removeEmbellishedDocument((String) depender.get(0), connection)
-            }
```

Comment:
Doing this inside the main transaction can lead to deadlocks even if we sort the dependencies since `doc` might also be part of the dependencies of another transaction.